src/os/filestore/chain_xattr.h

```diff
@@ -4,6 +4,7 @@
 #ifndef __CEPH_OSD_CHAIN_XATTR_H
 #define __CEPH_OSD_CHAIN_XATTR_H
 
+#include "include/compat.h"
```

Comment:
any specific reason you want to put `#include "include/compat.h"` before other includes? it seems we have a tradition to include system headers first, other 3rd party headers, and then headers in ceph project.